File: need4spd/euler_18.rb
triangle=[]

File.open("euler_18_problem.txt") do |file|
  while line=file.gets
    puts line
    r=line.scan(/\S+/).collect {|n| n.to_i}
    triangle.push(r)
  end
end

triangle = triangle.reverse

(0..13).each do |row|
  (0..13-row).each do |col|
    first_sum = triangle[row][col] + triangle[row+1][col]
    second_sum = triangle[row][col+1] + triangle[row+1][col]
    max_of_two = [first_sum, second_sum].max
    
    triangle[row+1][col] = max_of_two
  end
  
  triangle[row+1].each {|r| print r, " "}
  puts "\n"
end

File: need4spd/euler_18.py

sampleFile=open("euler_18_problem.txt", "r")

triangle=[]

for l in sampleFile.readlines():
    l=l.rstrip('\n')
    raw_list=l.split(" ")
    
    temp_l=list()
    for y in raw_list:
        temp_l.append(int(y))
    
    triangle.append(temp_l)
    
# reverse.
triangle.reverse();

# add each row to row+1 and replace row
for row in range(0,14):
    for col in range(0, 14-row):
        first_sum = triangle[row][col] + triangle[row+1][col]
        second_sum = triangle[row][col+1] + triangle[row+1][col]
        max_of_two = max(first_sum, second_sum)
        
        triangle[row+1][col] = max_of_two
        
    print("triangle : {0}".format(triangle[row+1]))
